hypershift-operator/controllers/nodepool/azure.go

```diff
@@ -51,6 +54,24 @@ func azureMachineTemplateSpec(hcluster *hyperv1.HostedCluster, nodePool *hyperv1
 		FailureDomain:          failureDomain(nodePool),
 	}}}
 
+	if nodePool.Spec.Platform.Azure.ImageID != "" {
+		azureMachineTemplate.Template.Spec.Image = &capiazure.Image{ID: ptr.To(nodePool.Spec.Platform.Azure.ImageID)}
+	}
+
+	if nodePool.Spec.Platform.Azure.MarketplaceImageInfo.Publisher != "" {
```

Comment:
does this make image marketplace takes precedence over image ID ?

Reply:
No image ID seems to take precedence. See the API where I added this MarketplaceImageInfo. I believe I added the context there.

Reply:
but if both are set then where will the azuremachinetemplate get the image.ID from if were overwriting it with the marketplaceImage down below?

Reply:
This should be fixed now